query-parser/src/parse.rs

```diff
@@ -119,6 +121,28 @@ def_parser!(Query, arguments, Vec<FnArg>, {
     (many::<Vec<FnArg>, _>(Query::fn_arg()))
 });
 
+def_parser!(Query, direction, Direction, {
+    satisfy_map(|v: edn::ValueAndSpan| {
```

Comment:
Huh, I don't care about this, but I would have expected this to have been `variable().optional(or("asc", desc"))`.  Where you worried about backtracing?